src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SVKmerizer.java

```diff
@@ -12,23 +12,23 @@
  * Iterator over successive Kmers from a sequence of characters.
  * Silently skips over parts of the sequence that has characters other than A, C, G, or T.
  */
-public class SVKmerizer implements Iterator<SVKmer> {
+public class SVKmerizer<KType extends SVKmerBase> implements Iterator<KType> {
```

Comment:
Done. The superclass is now SVKmer, extended by SVKmerShort and SVKmerLong. It required a lot more changes in some of the existing SV tools but I think results in overall cleaner code.